### ISSUE TITLE
CA1822 fires on a property whose setter accesses instance data

### ISSUE THREAD

**Issue by davidwengier** [dotnet/roslyn-analyzers#1933]
#### Analyzer package

[Microsoft.CodeQuality.Analyzers](https://www.nuget.org/packages/Microsoft.CodeQuality.Analyzers)

#### Package Version

v2.6.2

#### Diagnostic ID

[CA1822](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1822-mark-members-as-static?view=vs-2017)

#### Repro steps

```
    public class MyClass
    {
        private static bool StaticThing;
        private bool InstanceThing;

        public bool Thing
        {
            get { return StaticThing; }
            set
            {
                StaticThing = value;
                InstanceThing = value;
            }
        }
    }
```

#### Expected behavior

`Thing` is not flagged as the setter accesses instance data

#### Actual behavior

`Thing` is flagged as can be marked static
